core/src/main/java/org/apache/carbondata/core/datastore/chunk/impl/DimensionRawColumnChunk.java

```diff
@@ -126,4 +129,12 @@ public void setFileReader(FileReader fileReader) {
   public FileReader getFileReader() {
     return fileReader;
   }
+
+  public CarbonDictionary getLocalDictionary() {
+    return localDictionary;
+  }
+
+  public void setLocalDictionary(CarbonDictionary localDictionary) {
```

Comment:
Remove setter and use the only getter. Inside getter check null and uncompress it for first time read.